checker/src/main/java/org/checkerframework/checker/calledmethods/CalledMethodsTransfer.java

```diff
@@ -130,4 +139,90 @@ public void accumulate(Node node, TransferResult<CFValue, CFStore> result, Strin
+                annot,
+                ((CalledMethodsAnnotatedTypeFactory) atypeFactory)
+                    .ensuresCalledMethodsVarArgsValueElement,
+                String.class)
+            .toArray(new String[0]);
+    List<? extends VariableElement> parameters = elt.getParameters();
+    int varArgsPos = parameters.size() - 1;
+    Node varArgActual = node.getArguments().get(varArgsPos);
+    // In the CFG, explicit passing of multiple arguments in the varargs position is represented via
+    // an ArrayCreationNode.  This is the only case we handle for now.
+    if (varArgActual instanceof ArrayCreationNode) {
+      ArrayCreationNode arrayCreationNode = (ArrayCreationNode) varArgActual;
+      // add in the called method to all the vararg arguments
+      CFStore thenStore = result.getThenStore();
+      CFStore elseStore = result.getElseStore();
+      for (Node arg : arrayCreationNode.getInitializers()) {
+        AnnotatedTypeMirror currentType = atypeFactory.getAnnotatedType(arg.getTree());
+        AnnotationMirror newType = getUpdatedCalledMethodsType(currentType, ensuredMethodNames);
+        if (newType == null) {
+          continue;
+        }
+
+        JavaExpression receiverReceiver = JavaExpression.fromNode(arg);
+        thenStore.insertValue(receiverReceiver, newType);
+        elseStore.insertValue(receiverReceiver, newType);
+      }
+    }
+  }
+
+  /**
+   * Extract the current called-methods type from {@code currentType}, and then add each element of
+   * {@code methodNames} to it, and return the result. This method is similar to GLB, but should be
+   * used when the new methods come from a source other than an {@code CalledMethods} annotation.
+   *
+   * @param currentType the current type in the called-methods hierarchy
+   * @param methodNames the names of the new methods to add to the type
+   * @return the new annotation, to be added to the type
+   */
+  private AnnotationMirror getUpdatedCalledMethodsType(
+      AnnotatedTypeMirror currentType, String... methodNames) {
+    AnnotationMirror type;
+    if (currentType == null || !currentType.isAnnotatedInHierarchy(atypeFactory.top)) {
+      type = atypeFactory.top;
+    } else {
+      type = currentType.getAnnotationInHierarchy(atypeFactory.top);
+    }
+
+    // Don't attempt to strengthen @CalledMethodsPredicate annotations, because that would
+    // require reasoning about the predicate itself. Instead, start over from top.
+    if (AnnotationUtils.areSameByName(
+        type, "org.checkerframework.checker.calledmethods.qual.CalledMethodsPredicate")) {
+      type = atypeFactory.top;
+    }
+
+    if (AnnotationUtils.areSame(type, atypeFactory.bottom)) {
+      return null;
+    }
+
+    CalledMethodsAnnotatedTypeFactory cmAtf = (CalledMethodsAnnotatedTypeFactory) atypeFactory;
+
+    List<String> currentMethods =
+        AnnotationUtils.getElementValueArray(type, cmAtf.calledMethodsValueElement, String.class);
```

Comment:
I would put `calledMethodsValueElement` in a field of this class to avoid computing every time (mostly to clarify the code rather than for the efficiency benefit), but either way is OK.